src/components/text/docs.mdx

```diff
@@ -0,0 +1,29 @@
+---
+componentName: Text
+---
+
+The `Text` component should be used for easily styling `p` elements to conform to the design system.
+
+## Props
+
+In addition to the props described in the table below, this component also accepts any other valid props for heading elements. A caveat:
+
+- The given `className` prop is applied last to the list of class names set on the rendered element. This means that some CSS properties applied with the following props can be overwritten if needed (not recommended).
+
+| Prop name   | Description                                             | Required? | Default value | Type                                            |
+| ----------- | ------------------------------------------------------- | --------- | ------------- | ----------------------------------------------- |
+| `asElement` | The name of the element tag to render.                  | No        | "p"           | One of: "li", "p", "span"                       |
```

Comment:
for consideration: `as` is a very common naming for a prop which does this across many different libraries and systems. Would we consider going with `as` instead of `asElement`?

Reply:
I thought about using `as`, but I think `asElement` because is verbose and more clear.

Reply:
+1 on using `as` (I have left my comment above, so you can see I had the same suggestion even before reading this comment)

Reply:
@BRKalow @didoo are there any drawbacks to using `asElement`? I think it's still accessible to folks used to using `as` via autocomplete and I really value the explicit verbosity that the prop affects the element rendered.

Reply:
It's just additional cognitive overhead (for me personally) and more characters to type, but no I don't think are any material drawbacks. 😄  Happy to roll with `asElement` if you feel it's a marked improvement!

Reply:
Thank you @BRKalow! I do see it as an improvement. From my perspective the more verbose name directly benefits people who are newer to programming, new to this codebase, tired or struggling to focus, neurodivergent, and probably more. I've personally never seen `as` used before, and I would find it very confusing the first few times in code without the additional context.